From 3435a8758abb3db17600aa0dd10c1d1621fdb073 Mon Sep 17 00:00:00 2001
From: Zhao Qiang <qiang.zhao@nxp.com>
Date: Thu, 31 May 2018 10:55:45 +0800
Subject: [PATCH 532/767] usb: dwc3: revert initiated xhci probe problem in
 arm64 4.4 kernel due to DMA setup & fix panic

This reverts commit e5a00b9a04c91ff343d8fecf12c7e821e363bbbf.

Fix panic, log is as below.

[    4.100565] [<ffff000008a7f378>] of_dma_configure+0x12c/0x1d8
[    4.106302] [<ffff000008936a80>] dwc3_host_init+0x98/0x3c4
[    4.111778] [<ffff000008936894>] dwc3_probe+0xb14/0xba4
[    4.116994] [<ffff000008699690>] platform_drv_probe+0x50/0xbc
[    4.122729] [<ffff0000086978d4>] driver_probe_device+0x220/0x2dc
[    4.128724] [<ffff000008697a34>] __driver_attach+0xa4/0xa8
[    4.134198] [<ffff000008695a84>] bus_for_each_dev+0x58/0x98
[    4.139760] [<ffff0000086971f8>] driver_attach+0x20/0x28
[    4.145060] [<ffff000008696d78>] bus_add_driver+0x1c0/0x224
[    4.150621] [<ffff00000869833c>] driver_register+0x68/0x108
[    4.156182] [<ffff0000086995d4>] __platform_driver_register+0x48/0x50
[    4.162613] [<ffff00000921a054>] dwc3_driver_init+0x18/0x20
[    4.168176] [<ffff000008084144>] do_one_initcall+0x38/0x128
[    4.173738] [<ffff0000091d0d58>] kernel_init_freeable+0x190/0x230
[    4.179822] [<ffff000008ca7898>] kernel_init+0x10/0x100
[    4.185036] [<ffff000008085390>] ret_from_fork+0x10/0x18
Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/dwc3/host.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index c02216d61eda..0a3427452bf8 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -9,8 +9,6 @@
 
 #include <linux/platform_device.h>
 
-#include <linux/of_device.h>
-
 #include "core.h"
 
 static int dwc3_host_get_irq(struct dwc3 *dwc)
@@ -79,14 +77,7 @@ int dwc3_host_init(struct dwc3 *dwc)
 		return -ENOMEM;
 	}
 
-	if (IS_ENABLED(CONFIG_OF) && dwc->dev->of_node)
-		of_dma_configure(&xhci->dev, dwc->dev->of_node);
-	else
-		dma_set_coherent_mask(&xhci->dev, dwc->dev->coherent_dma_mask);
-
-	xhci->dev.parent = dwc->dev;
-
-	xhci->dev.dma_mask = dwc->dev->dma_mask;
+	xhci->dev.parent	= dwc->dev;
 
 	dwc->xhci = xhci;
 
-- 
2.17.0

